core/packages/PackageInfo.cc

```diff
@@ -467,26 +467,9 @@ optional<string> PackageInfo::pathTo(const core::GlobalState &gs, const MangledN
     return nullopt;
 }
 
-void PackageInfo::trackPackageReference(const core::FileRef file, const core::packages::MangledName package,
-                                        const PackageReferenceInfo packageReferenceInfo) {
-    auto r = referencedPackages.find(package);
-    if (r != referencedPackages.end()) {
-        r->second.first.insert(file);
-    } else {
-        referencedPackages[package] = {{file}, packageReferenceInfo};
-    }
-}
-
-void PackageInfo::untrackPackageReferencesFor(const core::FileRef file) {
-    for (auto &[_, v] : referencedPackages) {
-        auto files = v.first;
-        auto it = files.find(file);
-        if (it == files.end()) {
-            continue;
-        }
-        files.erase(it);
-    }
-    erase_if(referencedPackages, [](const auto &x) { return x.second.first.empty(); });
+void PackageInfo::trackPackageReferences(
+    const core::FileRef file, std::vector<std::pair<core::packages::MangledName, PackageReferenceInfo>> &references) {
```

Comment:
```suggestion
void PackageInfo::trackPackageReferences(FileRef file, vector<pair<MangledName, PackageReferenceInfo>> &references) {
```

you can get this to fit on one line and still compile if you'd like.